src/platform/datapath_winkernel.c

```diff
@@ -2923,6 +2973,18 @@ SocketSend(
         *(PINT)WSA_CMSG_DATA(CMsg) = SendData->ECN;
     }
 
+    if (Binding->Datapath->Features & CXPLAT_DATAPATH_FEATURE_DSCP) {
+        CMsg = (PWSACMSGHDR)&CMsgBuffer[CMsgLen];
+        CMsgLen += WSA_CMSG_SPACE(sizeof(INT));
+        CMsg->cmsg_level =
+            Route->LocalAddress.si_family == QUIC_ADDRESS_FAMILY_INET ?
```

Comment:
LocalAddress might be the wrong address to use; if trying to send to an IPv4 address using a dual-mode socket, you need to set the IPv4 IP_TOS, instead of the IPV6_TCLASS